alacritty_terminal/src/clipboard.rs

```diff
@@ -18,35 +18,47 @@ use std::ffi::c_void;
 use log::{debug, warn};
 
 use copypasta::nop_clipboard::NopClipboardContext;
-#[cfg(not(any(target_os = "macos", target_os = "windows")))]
+#[cfg(all(not(any(target_os = "macos", windows)), feature = "wayland"))]
 use copypasta::wayland_clipboard;
-#[cfg(not(any(target_os = "macos", target_os = "windows")))]
+#[cfg(all(not(any(target_os = "macos", windows)), feature = "x11"))]
 use copypasta::x11_clipboard::{Primary as X11SelectionClipboard, X11ClipboardContext};
-use copypasta::{ClipboardContext, ClipboardProvider};
+#[cfg(any(feature = "x11", target_os = "macos", windows))]
+use copypasta::ClipboardContext;
+use copypasta::ClipboardProvider;
 
 pub struct Clipboard {
     clipboard: Box<dyn ClipboardProvider>,
     selection: Option<Box<dyn ClipboardProvider>>,
 }
 
 impl Clipboard {
-    #[cfg(any(target_os = "macos", target_os = "windows"))]
+    #[cfg(any(target_os = "macos", windows))]
     pub fn new() -> Self {
         Self::default()
     }
 
-    #[cfg(not(any(target_os = "macos", target_os = "windows")))]
-    pub fn new(display: Option<*mut c_void>) -> Self {
-        if let Some(display) = display {
-            let (selection, clipboard) =
-                unsafe { wayland_clipboard::create_clipboards_from_external(display) };
-            return Self { clipboard: Box::new(clipboard), selection: Some(Box::new(selection)) };
+    #[cfg(not(any(target_os = "macos", windows)))]
+    pub fn new(_display: Option<*mut c_void>) -> Self {
```

Comment:
`display` should not be given an underscore, as the binding is in fact used. I find this confusing. I personally don't have an issue with shadowing in Rust however, so that's how I'd personally fix this, like it was before.

Reply:
The reason why an underscore is used, is because when you compile without the `Wayland` feature, the `display` variable will actually never be read.

I think the underscore is a perfectly fine solution in this case. It's just super nasty code because there are way too many feature flags involved.